src/backend/aspen/app/views/tests/test_usergroup_views.py

```diff
@@ -12,7 +12,12 @@ def test_usergroup_view_get(session, app, client):
     with client.session_transaction() as sess:
         sess["profile"] = {"name": user.name, "user_id": user.auth0_user_id}
     res = client.get("/api/usergroup")
-    expected = {"user": user.to_dict(), "group": group.to_dict()}
+    expected = {
+        "user": {
+            k: v for k, v in user.to_dict().items() if k in ["name", "agreed_to_tos"]
+        },
+        "group": {k: v for k, v in user.group.to_dict().items() if k in ["name"]},
+    }
```

Comment:
I would say that at this point, you might as well be explicit about what you're returning:

```
{"user": { "name": "xxx", "agreed_to_tos": "yyy"}, "group": {"name": "zzz"}}
```

Reply:
i made some changes where i wrap this in a function, maybe it's a better approach

Reply:
also i tried doing:
`session.query(User.name, User.agreed_to_tos, Group.name).select_from(User).join(User.Group)` and 
`session.query(User).options(load_only(“agreed_to_tos”, “name”), joinedload(User.group).load_only(“name”))` to do the filtering in the query itself, but both methods had faults